### PR TITLE
[v6r12] fix fts150114

### PR DESCRIPTION
**Pull request by phicharp**
Main change: return all FileCatalog errors
VOs using more than one catalog should treat this correctly. In LHCb we return S_OK() for actions on files starting with /lhcb/user

### CODE REVIEW

File: DataManagementSystem/Agent/FTSAgent.py [hunk 1]
```diff
@@ -925,7 +925,10 @@ def __finalizeFTSJob( self, request, ftsJob ):
     if not processFiles["OK"]:
       log.error( processFiles["Message"] )
       return processFiles
-    ftsFilesDict = self.updateFTSFileDict( ftsFilesDict, processFiles["Value"] )
+    processFiles = processFiles['Value']
+    if processFiles['toRegister']:
+      log.error( "%f files could not be registered in FC" % len( processFiles['toRegister'] ) )
```
Comment (atsareg) [inline DataManagementSystem/Agent/FTSAgent.py]: I remind your again that log error messages must have a static part and a variable part as the first and the second argument. So, this line should rather be:
log.error( "A number of files could not be registered in FC", len( processFiles['toRegister'] ) )

Comment (phicharp) [inline DataManagementSystem/Agent/FTSAgent.py]: This is indeed a non-existing requirement I guess… or a joke  ;-)))
I have been fixing hundreds of log calls with “dfsfd gssf g %s” % res[‘Message’] which I think makes sense, but reporting the number of items makes no sense in the way you write it, I am sorry… and it is totally useless unless the two items are reported separately in some magic DB somewhere???
If this is mandatory, don’t make the second argument optional… and fix all the calls throughout DIRAC…
Philippe

Le 19 janv. 2015 à 23:33, Andrei Tsaregorodtsev <notifications@github.com> a écrit :

> In DataManagementSystem/Agent/FTSAgent.py:
> 
> > @@ -925,7 +925,10 @@ def __finalizeFTSJob( self, request, ftsJob ):
> >      if not processFiles["OK"]:
> >        log.error( processFiles["Message"] )
> >        return processFiles
> > -    ftsFilesDict = self.updateFTSFileDict( ftsFilesDict, processFiles["Value"] )
> > +    processFiles = processFiles['Value']
> > +    if processFiles['toRegister']:
> > +      log.error( "%f files could not be registered in FC" % len( processFiles['toRegister'] ) )
> I remind your again that log error messages must have a static part and a variable part as the first and the second argument. So, this line should rather be:
> log.error( "A number of files could not be registered in FC", len( processFiles['toRegister'] ) )
> 
> —
> Reply to this email directly or view it on GitHub.
> 

-- 
Dr Philippe Charpentier
J09210, Physics Department, CERN, CH-1211 Genève 23
LHCb Experiment Distributed Computing Coordinator
Office 2-R-007, Tel : +41 22 767 4244 ,  GSM : +41 76 487 0167
mailto:Philippe.Charpentier@cern.ch  http://cern.ch/Philippe.Charpentier

Comment (atsareg) [inline DataManagementSystem/Agent/FTSAgent.py]: The Magic DB is the SystemLoggingDB where all the entries are classified by the static message part which is the first argument. The variable part is printed in the log, so you can print the number of files there. Make it so, that it makes sense for the reader and does not screw up SystemLogging at the same time.

Comment (phicharp) [inline DataManagementSystem/Agent/FTSAgent.py]: OK, but then every developer should go through the code as I can tell you that there are very many non-static first part!!!!
May be it is worth a reminder to all developers: every time you touch a module, fix the logging...

Le 20 janv. 2015 à 00:39, Andrei Tsaregorodtsev <notifications@github.com> a écrit :

> In DataManagementSystem/Agent/FTSAgent.py:
> 
> > @@ -925,7 +925,10 @@ def __finalizeFTSJob( self, request, ftsJob ):
> >      if not processFiles["OK"]:
> >        log.error( processFiles["Message"] )
> >        return processFiles
> > -    ftsFilesDict = self.updateFTSFileDict( ftsFilesDict, processFiles["Value"] )
> > +    processFiles = processFiles['Value']
> > +    if processFiles['toRegister']:
> > +      log.error( "%f files could not be registered in FC" % len( processFiles['toRegister'] ) )
> The Magic DB is the SystemLoggingDB where all the entries are classified by the static message part which is the first argument. The variable part is printed in the log, so you can print the number of files there. Make it so, that it makes sense for the reader and does not screw up SystemLogging at the same time.
> 
> —
> Reply to this email directly or view it on GitHub.
> 

-- 
Dr Philippe Charpentier
J09210, Physics Department, CERN, CH-1211 Genève 23
LHCb Experiment Distributed Computing Coordinator
Office 2-R-007, Tel : +41 22 767 4244 ,  GSM : +41 76 487 0167
mailto:Philippe.Charpentier@cern.ch  http://cern.ch/Philippe.Charpentier

Comment (phicharp) [inline DataManagementSystem/Agent/FTSAgent.py]: What can I do with this:
    for i, site in enumerate( self.__ftsGraph.nodes() ):
      log.debug( " [%02d] FTSSite: %-25s FTSServer: %s" % ( i, site.name, site.FTSServer ) )
    log.debug( "FTSRoutes: %s" % len( self.__ftsGraph.edges() ) )
    for i, route in enumerate( self.__ftsGraph.edges() ):
      log.debug( " [%02d] FTSRoute: %-25s Active FTSJobs (Max) = %s (%s)" % ( i,

Comment (phicharp) [inline DataManagementSystem/Agent/FTSAgent.py]: If you want to continue the "cleaning" of the FTSAgent, please go ahead, but I gave up, much more important things to do...

**Pull request merged**